packages/@react-aria/dnd/docs/useClipboard.mdx

```diff
@@ -0,0 +1,432 @@
+
+The example below works with the above `Copyable` example using a custom app-specific data format to transfer rich data. If no such data is available, it falls back to pasting plain text data.
+
+```tsx example
+///- begin collapse -///
+function Copyable() {
+  let {clipboardProps} = useClipboard({
+    getItems() {
+      return [{
+        'text/plain': 'hello world',
+        'text/html': '<strong>hello world</strong>',
+        'my-app-custom-type': JSON.stringify({
+          message: 'hello world',
+          style: 'bold'
+        })
+      }, {
+        'text/plain': 'foo bar',
+        'text/html': '<em>foo bar</em>',
+        'my-app-custom-type': JSON.stringify({
+          message: 'foo bar',
+          style: 'italic'
+        })
+      }];
+    }
+  });
+
+  return (
+    <div role="textbox" tabIndex={0} {...clipboardProps}>
+      <div>
+        <div><strong>hello world</strong></div>
+        <div><em>foo bar</em></div>
+      </div>
+      <kbd>⌘C</kbd>
+    </div>
+  );
+}
+///- end collapse -///
+function Pasteable() {
+  let [pasted, setPasted] = React.useState(null);
+  let {clipboardProps} = useClipboard({
+    async onPaste(items) {
+      let pasted = await Promise.all(
+        items
+          .filter(item => item.kind === 'text' && (item.types.has('text/plain') || item.types.has('my-app-custom-type')))
+          .map(async item => {
+            if (item.types.has('my-app-custom-type')) {
+              return JSON.parse(await item.getText('my-app-custom-type'));
+            } else {
+              return {message: await item.getText('text/plain')};
+            }
+          })
+      );
+      setPasted(pasted);
+    }
+  });
+
+  let message = ['Paste here'];
+  if (pasted) {
+    message = pasted.map(d => {
+      let message = d.message;
+      if (d.style === 'bold') {
+        message = <strong>{message}</strong>;
+      } else if (d.style === 'italic') {
+        message = <em>{message}</em>;
+      }
+      return <div>{message}</div>;
+    });
+  }
+
+  return (
+    <div role="textbox" tabIndex={0} {...clipboardProps}>
+      <div>{message || 'Paste here'}</div>
+      <kbd>⌘V</kbd>
+    </div>
+  );
+}
+///- begin collapse -///
+<Copyable />
+<Pasteable />
+///- end collapse -///
+```
+
+### Files
+
+A <TypeLink links={sharedDocs.links} type={sharedDocs.links[sharedDocs.exports.FileItem.id]} /> references a file on the user's device. It includes the name and mime type of the file, and methods to read the contents as plain text, or retrieve a native [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object which can be attached to form data for uploading.
```

Comment:
Same w/ FileItem and DirectoryItem, we'll need to add some descriptions to the properties/methods